services/src/main/java/org/keycloak/connections/httpclient/DefaultHttpClientFactory.java

```diff
@@ -107,20 +108,14 @@ public int postText(String uri, String text) throws IOException {
             }
 
             @Override
-            public InputStream get(String uri) throws IOException {
+            public String get(String uri) throws IOException {
                 HttpGet request = new HttpGet(uri);
                 HttpResponse response = httpClient.execute(request);
-                int statusCode = response.getStatusLine().getStatusCode();
-                HttpEntity entity = response.getEntity();
-                if (statusCode < 200 || statusCode >= 300) {
-                    EntityUtils.consumeQuietly(entity);
-                    throw new IOException("Unexpected HTTP status code " + response.getStatusLine().getStatusCode() + " when expecting 2xx");
-                }
-                if (entity == null) {
+                String body = responseHandler.handleResponse(response);
```

Comment:
I would have preferred to use `EntityUtils.toString(response.getEntity())` which is what this handler is doing in the end. Just because it would be easier to be changed in the future. But just nitpicking. It's approved anyway.

Reply:
I didn't want the error handling to be duplicated which is already present in the `AbstractResponseHandler`. Still I agree it is a matter of how much changes we would expect here in the future.